add-on/panels/js/sidebar.jsx

```diff
@@ -119,17 +119,34 @@ loop.sidebar = (function(mozL10n) {
         }
       });
 
+      // Plug in an alternate client ID mechanism, as localStorage and cookies
+      // don't work in the conversation window
+      var currGuid = results[++requestIdx];
+      window.OT.overrideGuidStorage({
+        get: function(callback) {
+          callback(null, currGuid);
+        },
+        set: function(guid, callback) {
+          // See nsIPrefBranch
+          var PREF_STRING = 32;
+          currGuid = guid;
+          loop.request("SetLoopPref", "ot.guid", guid, PREF_STRING);
+          callback(null);
+        }
+      });
```

Comment:
Yes we do need to keep this (desktop only). You could probably change the comment to end with something like "don't work in about: pages".